### DISCUSSION TITLE
Difference between `try` and `sequence`

### DISCUSSION THREAD

**Discussion by smachave** [mint-lang/mint#410] [Q&A] [answered]
I've read the documentation for the two several times, and the differences that I've spotted are:

1. `try` does not have a `finally` block
2. `next` is not used within a `try` block

How far am I right, and are there any more differences?

**Comment by gdotdesign** [comment, marked as answer]
Hi :wave: 

>     1. `try` does not have a `finally` block

That is correct and it's because `try` is synchronous.

>     2. `next` is not used within a `try` block

It can be but it won't be awaited so accessing a state right after a `next` call will return the previous value.

> How far am I right, and are there any more differences?

The fundamental difference is that `try` is synchronous whereas `sequence` is asynchronous. In `try` results (`Result(error, value)`) are unboxed, in `sequence` promises (`Promise(error, value)`) are awaited and unboxed as well.

**Reply by smachave** [reply]
Thank you for the well worded answer.

**Comment by Sija** [comment]
No longer relevant, as these constructs were removed in v0.17.0.
